src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,123 @@
+package duke;
+
+import java.util.Scanner;
+
+/**
+ * Deals with interactions with the user.
+ */
+public class Ui {
+
+    private Scanner input = new Scanner(System.in);
+
+    /**
+     * Obtains the user input.
+     *
+     * @return String representation of the user input.
+     */
+    public String getInput() {
+        return input.nextLine();
+    }
+
+    /**
+     * Greets the user when the chatbot is started.
+     */
+    public void printGreeting() {
+        String greet = "Hi! I'm Uke\n" + "What can I do for you?\n";
+        System.out.println(greet);
+    }
+
+    /**
+     * Prints the list of tasks in the TaskList given.
+     *
+     * @param storedTasks TaskList to be printed.
+     * @param isListCommand Whether the method is called from a list command.
+     */
+    public void printList(TaskList storedTasks, boolean isListCommand) {
+        int len = storedTasks.getLength();
+
+        if (isListCommand) {
+            if (len > 0) {
```

Comment:
Perhaps can add this as a guard clause at the start of the method? Since the function can terminate if the list is empty 🤔 
For example:

```java
if (len == 0) {
    System.out.println("No tasks found!");
    return; // terminates the method early since there is nothing to print
}
```